src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -36,117 +32,137 @@
 
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.security.KeyPair;
+import java.security.KeyStoreException;
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
-import java.time.Clock;
+import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 
-import static com.aws.greengrass.clientdevices.auth.ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME;
 import static com.aws.greengrass.clientdevices.auth.configuration.CAConfiguration.CA_CERTIFICATE_URI;
 import static com.aws.greengrass.clientdevices.auth.configuration.CAConfiguration.CA_PRIVATE_KEY_URI;
 import static com.aws.greengrass.clientdevices.auth.configuration.CAConfiguration.CERTIFICATE_AUTHORITY_TOPIC;
 import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @ExtendWith({MockitoExtension.class, GGExtension.class})
 public class CustomCaConfigurationTest {
     @Mock
-    private SessionManager sessionManagerMock;
-
+    SecurityService securityServiceMock;
     @Mock
-    private GroupManager groupManagerMock;
-
-    @Mock private CertificateRegistry certificateRegistryMock;
-
+    private GreengrassServiceClientFactory clientFactory;
     @Mock
-    ConnectivityInformation mockConnectivityInformation;
-
+    private GroupManager groupManager;
     @Mock
-    CertificateExpiryMonitor mockCertExpiryMonitor;
-
+    CertificateExpiryMonitor certExpiryMonitor;
     @Mock
-    CISShadowMonitor mockShadowMonitor;
-
-    @Mock
-    GreengrassServiceClientFactory clientFactoryMock;
-    @Mock
-    SecurityService securityServiceMock;
+    private GreengrassV2DataClient client;
     @TempDir
-    Path tmpPath;
-
-    private CertificateManager certificateManager;
-    private CertificateStore certificateStore;
+    Path rootDir;
+    private Kernel kernel;
 
 
     @BeforeEach
-    void beforeEach() {
-        DomainEvents events = new DomainEvents();
-        certificateStore = new CertificateStore(tmpPath, events);
```

Comment:
I refactored this tests so they are more integration like instead of calling direct units.